File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        s=[]
        def traverse(root):
            if root is not None:
                s.append(root.val)
                traverse(root.left)
                traverse(root.right)
        traverse(root)
        s.sort()
        return s[k-1]

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void in(TreeNode*root,vector<int>&v){
        if(root==nullptr)return;
        in(root->left,v);
        v.push_back(root->val);
        in(root->right,v);
    }
    int kthSmallest(TreeNode* root, int k) {
        vector<int> v;
    in(root,v);
    sort(v.begin(),v.end());
    return v[k-1];
    }
};